### DISCUSSION TITLE
Figure out which action triggered the StateChange?

### DISCUSSION THREAD

**Discussion by Nickztar** [mrpmorris/Fluxor#418] [Q&A]
Hi, so we currently have a large application with multiple states where some actions are sent multiple times a second to keep the state up to date which triggered a lot of StateChanges. What I would like to do is be able to ignore these changes by checking if the action that triggered the change was the most frequent action since it's mostly sent to be able to keep state up to date.

Tried to check if the Sender or EventArgs on the StateChange event had any information about this but did seem to be the case. Maybe this isn't the correct way to solve the problem but it's my current solution. Just wanted to make sure that im not missing anything here before I try to go down another road.

**Comment by mrpmorris** [comment]
You are explaining how you are trying to achieve something.

Instead, tell me what user experience you are trying to achieve and I'll see if I can help.

**Comment by Nickztar** [comment]
I would like to reduce the amount of StateHasChanged that is triggered for each message. Currently for every action we get a StateHasChanged which slows down our large lists severly. What I would like is to check the EventArgs or Sender in the StateChanged event and see which action triggered it and then prevent triggering StateHasChanged. 

Managed to figure out by adding a boolean in the state which gets set when the reducer for this action runs, then I can check for it and decide what to do.

**Comment by mrpmorris** [comment]
Sounds to me like you should be using `IStateSelection<TState, TSubset>`?
